cmd/monitoring.go

```diff
@@ -0,0 +1,387 @@
+	var name string
+	stream.FromArray(rules).Map(func(rule *common.RepositoryContent) string {
+		name = *rule.Name
+		content, err := service.DownloadContents(rule)
+		common.CheckErr(err, "")
+
+		if content == nil {
+			return ""
+		}
+
+		return string(content)
+	}).Filter(func(content string) bool {
+		return content != ""
+	}).Peek(func(content string) {
+		// ansible
+		common.WriteFile(ansibleRuleDir, name, content)
+	}).Each(func(content string) {
+		// operatotr
+		operator.WriteRule(content, name, operatorRuleDir, operatorReplaceExpr)
+	})
+}
+
+func getAnsibleDir(baseTagDir string) string {
+	return fmt.Sprintf("%s%c%s", baseTagDir, filepath.Separator, Ansible)
+}
+
+func getOperatorDir(baseTagDir string) string {
+	return fmt.Sprintf("%s%c%s", baseTagDir, filepath.Separator, Opertaor)
+}
+
+func RepoService(cfg *Config) (*common.GitRepoService, error) {
+	if cfg.Token != "" {
+		return common.NewGitRepoServiceWithAuth(common.BasicAuthTransport{
+			OTP: cfg.Token,
+		})
+	}
+
+	if cfg.UserName != "" || cfg.Password == "" {
+		return common.NewGitRepoServiceWithAuth(common.BasicAuthTransport{
+			Username: cfg.UserName,
+			Password: cfg.Password,
+		})
+	}
+
+	return common.NewGitRepoService()
+}
+
+func copyOperatorLocalfiles(sourcePath string, dstPath string) {
+	operatorConfig := fmt.Sprintf("%s%c%s", getPlatFormConfigDir(), filepath.Separator, Opertaor)
+	files := common.ListAllFiles(fmt.Sprintf("%s%c%s", operatorConfig, filepath.Separator, sourcePath))
+
+	stream.FromArray(files).Each(func(file string) {
+		df, err := ioutil.ReadFile(file)
+		common.CheckErr(err, fmt.Sprintf("read file failed, file=%s", file))
+
+		dstDir := fmt.Sprintf("%s%c%s%c%s", baseTagDir, filepath.Separator, Opertaor, filepath.Separator, dstPath)
+		if !common.PathExist(dstDir) {
+			os.MkdirAll(dstDir, os.ModePerm)
+		}
+		common.CheckErr(ioutil.WriteFile(fmt.Sprintf("%s%c%s", dstDir, filepath.Separator, common.ExtractFromPath(file)), df, os.ModePerm), "create file failed")
+	})
+}
+
+func copyAnsibleLocalFiles(sourcePath string, dstPath string) {
+	ansibleConfig := fmt.Sprintf("%s%c%s", getPlatFormConfigDir(), filepath.Separator, Ansible)
+
+	files := common.ListAllFiles(fmt.Sprintf("%s%c%s", ansibleConfig, filepath.Separator, sourcePath))
+
+	stream.FromArray(files).Each(func(file string) {
+		df, err := ioutil.ReadFile(file)
+		common.CheckErr(err, fmt.Sprintf("read file failed, file=%s", file))
+
+		dstDir := fmt.Sprintf("%s%c%s%c%s", baseTagDir, filepath.Separator, Ansible, filepath.Separator, dstPath)
+		if !common.PathExist(dstDir) {
+			os.MkdirAll(dstDir, os.ModePerm)
+		}
+		common.CheckErr(ioutil.WriteFile(fmt.Sprintf("%s%c%s", dstDir, filepath.Separator, common.ExtractFromPath(file)), df, os.ModePerm), "create file failed")
+	})
+}
+
+func getPlatFormConfigDir() string {
+	return removeLastSlash(platformMonitoringDir)
+}
+
+func removeLastSlash(str string) string {
```

Comment:
Can you just use [`strings.TrimRight`](https://golang.org/pkg/strings/#TrimRight)

Reply:
Good idea.